synapse/http/client.py

```diff
@@ -246,6 +254,15 @@ def get_raw(self, uri, args={}):
         else:
             raise CodeMessageException(response.code, body)
 
+    def _exceptionFromFailedRequest(self, response, body):
+        try:
+            jsonBody = json.loads(body)
+            errcode = jsonBody['errcode']
+            error = jsonBody['error']
+            return MatrixCodeMessageException(response.code, error, errcode)
+        except:
```

Comment:
Generally we prefer to list the types of exceptions, as otherwise its too easy to catch a typo or something. Its probably fine for me.